packages/cli/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt

```diff
@@ -275,27 +286,19 @@ import org.graalvm.polyglot.Engine as VMEngine
     )
     internal var sources: List<String> = emptyList()
 
-    /** Apply configuration to the VM based on the provided arguments. */
-    internal fun apply(debug: Boolean): Stream<VMProperty> {
-      return (if (!debug) emptyList<VMProperty>() else listOfNotNull(
-        // if specified, add custom `path`
-        when (val p = path) {
-          null -> null
-          else -> VMStaticProperty.of("inspect.Path", p)
-        },
-      ).plus(sources.joinToString(",").let { targets ->
-        // format + add any custom source directories
-        if (targets.isEmpty()) emptyList() else listOf(
-          VMStaticProperty.of("inspect.SourcePath", targets)
-        )
-      })).stream()
+    /** Apply these settings to the root engine configuration container. */
+    internal fun apply(config: PolyglotEngineConfiguration) {
+      // install and configure the Debug plugin
+      config.debug {
+        sourcePaths = sources
+        waitAttached = wait
+        path = this@DebugConfig.path
+        suspend = this@DebugConfig.suspend
+      }
     }
   }
 
-  internal sealed class LanguageSettings {
-    /** Apply configuration to the JS VM based on the provided arguments. */
-    abstract fun apply(): Stream<out VMProperty>
```

Comment:
Similar to `DebugConfig`, now each language settings container can be applied to the specific plugin configuration class (see `JavaScriptSettings` for an example).